extras/parse_travis_log.py

```diff
@@ -191,13 +191,41 @@ def print_static_analysis(d):
                 print(INDENT + ' | ' + 2 * INDENT +
                       ' - ' + k3 + ' : ' + str(v3))
 
+def count_warnings_errors(f):
+    """
+    Counts compiler warnings and errors. Stops when reading '+make install'.
+    """
+    warn = {}
+    error = {}
+    while True:
+        line = f.readline()
+        if not line:
+            return warn, error, line
+
+        if line.strip() == '+make install':
+            return warn, error, line
+
+        # /home/travis/build/tammoippen/nest-simulator/build/libnestutil/config.h:65:0: warning: "HAVE_DIRENT_H" redefined [enabled by default]
+        if ': warning:' in line:
+            file = line.split(':')[0]
```

Comment:
i'm fairly unhappy about the choice of using `file` as a variable name here, as this is also a Python object. I would use `file_name` instead.